### ISSUE TITLE
Reports discrepancy between "Sales by Date" and "Sales by Product" reports

### ISSUE THREAD

**Issue by finance-guy** [woocommerce/woocommerce#20541]
**Describe the bug**
If I pull a "Sales by Date" report for a particular period and look at the totals on the left, I can't match any of them to the sum of the top sellers or top earners list in the "Sales by Product" report for the same period. 

**To Reproduce**
1. Go to 'Sales Reports -> Orders -> Sales by Date'
2. Select the date range and click on "Go"
3. Note the Gross and Net Sales amounts
4. Go to 'Sales Reports -> Orders -> Sales by Product'
5. Select the date range and click on "Go"
6. Sum the amounts of the individual top sellers in the list or the top earners
7. I cannot match this total to either of the numbers in point 3 above.  

**Expected behavior**
The sum of the individual product sales should match the total sales figures on the Sales by Date report. It is closest to the net sales amount, so I'm assuming that's the number it should match. But it is currently lower than the total net sales. 

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by mikejolley** [comment]
tldr; They are different data sets and are unlikely to match exactly.

Sales by date reports at order level, so this also includes costs such as shipping, fees, and any manual adjustments you may have made to order totals. It also factors in any partial refunds.

Sales by product is gross; it reports on the line items within orders. This will not include shipping costs, fees, manual adjustments or refunds made on the order itself (unless the refund was made to the line).

Thats where the discrepancies will be.

If you don't do shipping, or use fees, but have done refunds, ensure that when you do the refund you mark it on the line items (by inputting cost and qty refunded per line) so this refund amount is attributed to the product/line item, not the order. Thats a common scenario which can affect reporting.

Hope this tracks down your issue :)

**Comment by sangemaru** [comment]
Hi, this still doesn't make sense. On our website, the sales by product for the year adds up to LESS than the net value reported under the sales by date report.
We took care to make all refunds to the lines, not to the order as a whole. The net sales for the sales by date report should also not be including the shipping costs and fees, so you'd expect these to match up, but we have about a $4k difference.

**Comment by finance-guy** [comment]
I have to agree, it doesn't make sense to me either still. We don't have shipping or fees on our transactions, only taxes on certain sales. Our per product sales total is less than the "Net Sales" on the Sales by Date report. Same as @sangemaru

**Comment by mikejolley** [comment]
Narrow the results down and look at the orders in that range - might show you more?

**Comment by sangemaru** [comment]
I ran some more reports across a variety of periods:

jan 1 - apr 30
net sales 49338.96
sales by product: 46137.63

--

jan 1 - jan 31
net sales 25815.73
sales by product 23757.7

--

jan 1 - jan 7 (these were almost identical, so i moved on to the next week)
810.19
810.18

--

jan 8 - Jan 14
net sales by date report:
9193.43
manual computation, excluding refunds and shipping, but NOT TAXES
8345.9
sales by product report:
8311.42

manually counted refunds - $882
NOTE that sales by date minus refunds = sales by product report for this period (which should not happen, as sales by date should include refunds)

---

The above data on the 2nd week of jan could be due to the fact that I went all the way through the orders for the year and modified all the refunds to be be line item instead of by order, which for reporting purposes appears to have moved nearly all of them into the month of june as far as the sales by product and sales by date reports are concerned, which shows up in the june data:

-----

june:
net sales by date:
$3665.65

sales by product:
$2624.02

NET manual counted sales:
8030.95
